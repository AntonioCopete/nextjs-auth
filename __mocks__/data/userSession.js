export const userSessionMock = {
	name: 'Username',
	email: 'email@domain.com',
	image: 'https://image-url',
};
