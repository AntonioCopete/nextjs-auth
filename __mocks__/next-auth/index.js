export const getServerSessionMock = jest.fn();

export const getServerSession = getServerSessionMock;

export default jest.fn();
