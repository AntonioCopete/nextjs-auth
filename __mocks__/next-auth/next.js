const NextAuthMock = jest.fn();

const NextAuth = NextAuthMock;
export default NextAuth;
