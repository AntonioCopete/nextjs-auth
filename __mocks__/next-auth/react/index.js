const signInMock = jest.fn();
const SessionProviderMock = jest.fn();
const useSessionMock = jest.fn();

export const signIn = signInMock;
export const SessionProvider = SessionProviderMock;
export const useSession = useSessionMock;
