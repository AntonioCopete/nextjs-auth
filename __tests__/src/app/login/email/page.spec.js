import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { signIn } from 'next-auth/react';
import LoginEmail from '../../../../../src/app/login/email/page';
import { userSessionMock } from '../../../../../__mocks__/data/userSession';

describe('Register', () => {
	beforeEach(() => {
		signIn.mockReset();
	});
	it('should match snapshot', () => {
		const component = render(<LoginEmail />);
		expect(component).toMatchSnapshot();
	});

	it('should login successfully', (done) => {
		signIn.mockReturnValueOnce({ user: userSessionMock });
		const component = render(<LoginEmail />);

		const emailInput = screen.getByPlaceholderText('email@gmail.com');
		const passwordInput = screen.getByPlaceholderText('********');
		fireEvent.change(emailInput, {
			target: { value: 'newemail@gmail.com' },
		});

		fireEvent.change(passwordInput, {
			target: { value: 'newpassword' },
		});

		const submitButton = screen.getByText('Login');
		fireEvent.click(submitButton);

		setTimeout(() => {
			expect(emailInput.value).toBe('newemail@gmail.com');
			expect(passwordInput.value).toBe('newpassword');

			expect(signIn.mock.calls[0][0]).toBe('login');
			expect(signIn.mock.calls[0][1].email).toBe(emailInput.value);
			expect(signIn.mock.calls[0][1].password).toBe(passwordInput.value);
			expect(signIn).toHaveBeenCalled();
			expect(component).toMatchSnapshot();
			done();
		}, 300);
	});

	it('should NOT login successfully', (done) => {
		signIn.mockReturnValueOnce({ error: 'error' });
		const component = render(<LoginEmail />);

		const submitButton = screen.getByText('Login');
		fireEvent.click(submitButton);

		setTimeout(() => {
			expect(signIn).toHaveBeenCalled();
			expect(component).toMatchSnapshot();
			done();
		}, 300);
	});

	it('should fail on signIn', (done) => {
		signIn.mockRejectedValue(new Error('signIn error'));
		const component = render(<LoginEmail />);

		const submitButton = screen.getByText('Login');
		fireEvent.click(submitButton);

		setTimeout(() => {
			expect(signIn).toHaveBeenCalled();
			expect(component).toMatchSnapshot();
			done();
		}, 300);
	});
});
