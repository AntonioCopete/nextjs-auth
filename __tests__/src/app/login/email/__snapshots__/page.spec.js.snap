// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should NOT login successfully 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="form"
      >
        <h1>
          Login with email
        </h1>
        <section>
          <label>
            Email:
            <input
              name="email"
              placeholder="email@gmail.com"
              type="email"
            />
          </label>
          <label>
            Password:
            <input
              name="password"
              placeholder="********"
              type="password"
            />
          </label>
          error
          <button
            type="submit"
          >
            Login
          </button>
        </section>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="form"
    >
      <h1>
        Login with email
      </h1>
      <section>
        <label>
          Email:
          <input
            name="email"
            placeholder="email@gmail.com"
            type="email"
          />
        </label>
        <label>
          Password:
          <input
            name="password"
            placeholder="********"
            type="password"
          />
        </label>
        error
        <button
          type="submit"
        >
          Login
        </button>
      </section>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Register should fail on signIn 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="form"
      >
        <h1>
          Login with email
        </h1>
        <section>
          <label>
            Email:
            <input
              name="email"
              placeholder="email@gmail.com"
              type="email"
            />
          </label>
          <label>
            Password:
            <input
              name="password"
              placeholder="********"
              type="password"
            />
          </label>
          <button
            type="submit"
          >
            Login
          </button>
        </section>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="form"
    >
      <h1>
        Login with email
      </h1>
      <section>
        <label>
          Email:
          <input
            name="email"
            placeholder="email@gmail.com"
            type="email"
          />
        </label>
        <label>
          Password:
          <input
            name="password"
            placeholder="********"
            type="password"
          />
        </label>
        <button
          type="submit"
        >
          Login
        </button>
      </section>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Register should login successfully 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="form"
      >
        <h1>
          Login with email
        </h1>
        <section>
          <label>
            Email:
            <input
              name="email"
              placeholder="email@gmail.com"
              type="email"
            />
          </label>
          <label>
            Password:
            <input
              name="password"
              placeholder="********"
              type="password"
            />
          </label>
          <button
            type="submit"
          >
            Login
          </button>
        </section>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="form"
    >
      <h1>
        Login with email
      </h1>
      <section>
        <label>
          Email:
          <input
            name="email"
            placeholder="email@gmail.com"
            type="email"
          />
        </label>
        <label>
          Password:
          <input
            name="password"
            placeholder="********"
            type="password"
          />
        </label>
        <button
          type="submit"
        >
          Login
        </button>
      </section>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Register should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="form"
      >
        <h1>
          Login with email
        </h1>
        <section>
          <label>
            Email:
            <input
              name="email"
              placeholder="email@gmail.com"
              type="email"
            />
          </label>
          <label>
            Password:
            <input
              name="password"
              placeholder="********"
              type="password"
            />
          </label>
          <button
            type="submit"
          >
            Login
          </button>
        </section>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="form"
    >
      <h1>
        Login with email
      </h1>
      <section>
        <label>
          Email:
          <input
            name="email"
            placeholder="email@gmail.com"
            type="email"
          />
        </label>
        <label>
          Password:
          <input
            name="password"
            placeholder="********"
            type="password"
          />
        </label>
        <button
          type="submit"
        >
          Login
        </button>
      </section>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
